### DISCUSSION TITLE
Plumbing/right click functionality on Windows?

### DISCUSSION THREAD

**Discussion by sqweek** [rjkroege/edwood#498] [Ideas]
Hey! I'm setting up a new windows laptop and looking for alternatives to the aging acme-sac, edwood is super cool thank you!

Right now it seems like right-click is tied solely to the Look function, eg. it won't even open URLs when I click on or highlight them, just searches within the buffer. One of my favourite acme tricks is using shorthand like "bug 23404" or "#5942" to refer to bugzilla issues/support tickets/etc. and then teaching the plumber how to interpret these strings and open the relevant ticket.

Do you have a feel for what would be required to get something like this functional on windows? I can see that look.go/look_posix.go does include plumber references so I'm wondering if it's up to me to find/port a windows plumber and run it alongside edwood?

**Comment by rjkroege** [comment]
Right click without plumb should:

* open files
* search in the buffer

This work for me on Windows. (My current job is forcing Windows use so I hope to finally spend some time on Edwood Windows issues.)

I think everything else (i.e. URLs) needs the plumber. I don't think that plumber works on Windows. I was going to run it in `wsl`. (I'm writing a tool that's suppose to let Edwood connect to a `plumber` running inside `wsl`.)

The "nice" way to support the plumber might be to write a Go lang `plumber` that runs on Windows? But I mostly wanted to plumb access to stuff inside my `wsl` VM so was exploring the connecting to an `wsl` vm.

**Reply by sfyatee** [reply]
Paul Lalonde has ported plumber to golang https://github.com/paul-lalonde/plumber. I have not tried it on Windows myself.

**Comment by sqweek** [comment]
Thanks for the info! I picked a bad time to start this conversation as I'm about to go on holiday for two months and have been busy preparing for that. I'm not taking a laptop with me but I anticipate having some time to spend on edwood in Windows when I return :)

**Comment by paul-lalonde** [comment]
It's far from complete.  I need to spend some time on it but summer keeps
interfering.

Paul
